fri/src/prover.rs

```diff
@@ -145,9 +145,9 @@ fn commit_phase<FC: FriConfig>(
         let beta: FC::Challenge = challenger.sample_ext_element();
         current = fold_even_odd(&current, beta);
 
-        let matrices = matrices_with_log_height(log_folded_height);
+        let matrices = &matrices_by_log_height[log_folded_height];
         if !matrices.is_empty() {
-            current = reduce_matrices(folded_height, &current, &matrices, alpha);
+            alpha_reducer.reduce_matrices(&mut current, folded_height, matrices);
```

Comment:
Hm previously each round used its own powers of `alpha` starting with `alpha^0`, but here it seems `alpha_reducer`'s previous `current_alpha_pow` state carries over, leading to higher `alpha` exponents in later rounds?

I think both variants should be sound (with a negligible error difference), but maybe restarting with `alpha^0` would simplify the API a little, since `MatrixReducer` could become stateless?

Reply:
ahh I missed that, maybe that’s why I was having trouble in verifier